pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -226,6 +229,19 @@ func init() {
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
 func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	vs.NodeManager().kubeClient = clientBuilder.ClientOrDie("vsphere-cloud-provider")
+
+	// Only on controller node it is required to register listeners.
+	// Register callbacks for node updates
+	// EventHandler needs to be added once kubeClient is provided to nodemanager
+	if vs.informerFactory != nil {
```

Comment:
Why not keep the earlier logic where node events methods are registered in SetInformers?

Reply:
Thanks for your review! What happens if we don't do this in Initialize method is that NodeAdded was called before kubeClient is passed to NodeManager, and NodeManager couldn't access configmaps.
We can keep it in SetInformers by adding some retry code in case that kubeClient is not yet to be given, but I feel it's more simple just to make sure NodeAdded/Deleted always called after kubeClient is provided instead. (Also I added L236 comment)